graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/PluginUtils.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.ballerina.stdlib.graphql.compiler;
+
+import io.ballerina.compiler.api.SemanticModel;
+import io.ballerina.compiler.api.symbols.MethodSymbol;
+import io.ballerina.compiler.api.symbols.ModuleSymbol;
+import io.ballerina.compiler.api.symbols.Qualifier;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.syntax.tree.FunctionDefinitionNode;
+import io.ballerina.projects.plugins.SyntaxNodeAnalysisContext;
+import io.ballerina.tools.diagnostics.Diagnostic;
+import io.ballerina.tools.diagnostics.DiagnosticFactory;
+import io.ballerina.tools.diagnostics.DiagnosticInfo;
+import io.ballerina.tools.diagnostics.DiagnosticSeverity;
+import io.ballerina.tools.diagnostics.Location;
+
+import java.util.Optional;
+
+/**
+ * Util class for the compiler plugin.
+ */
+public class PluginUtils {
+    public static Diagnostic getDiagnostic(PluginConstants.CompilationErrors error,
```

Comment:
It would be better you could write a utility method to update the context rather than getting the diagnostic and update the context manually each time. You could find a sample [here](https://github.com/ballerina-platform/module-ballerina-websub/blob/de891798d26fa8f4015500699cbe4ec77d17bebc/websub-compiler-plugin/src/main/java/io/ballerina/stdlib/websub/task/AnalyserUtils.java#L40).